.travis.yml

```diff
@@ -0,0 +1,64 @@
+#
+# Turn off email notifications entirely
+#
+notifications:
+  email: false
+
+language: generic
+
+#
+# Define the build matrix
+#
+matrix:
+  include:
+#    - services: docker
+#      env: OS_RELEASE=debian_unstable
+#      sudo: required
+#      script:
+#        - docker-compose -f scripts/docker/docker-compose.yml build $OS_RELEASE
+#        - docker-compose -f scripts/docker/docker-compose.yml run $OS_RELEASE /bin/bash /workspace/code/scripts/travis.sh
+    - os: osx
+
+osx_image: xcode7.3 # use latest OS X
+
+#
+# Some of the OS X images don't have cmake, contrary to what people
+# on the Internet say
+#
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      brew update &&
+      which cmake || brew install cmake &&
+      which swig || brew install swig &&
+      which python3 || brew install python3 &&
+      which lua || brew install lua &&
+      brew install gobject-introspection &&
+      brew install yajl &&
+      brew install augeas &&
+      brew cask install java;
+    fi
+#
+# Source is checked out in $TRAVIS_BUILD_DIR
+# Build directory will be TRAVIS_BUILD_DIR/../build
```

Comment:
I would prefer if you assign some variables for this, maybe some ROOT_DIR, BUILD_DIR, SOURCE_DIR and so on.